content/troubleshoot/docs/pvc-permission.md

```diff
@@ -0,0 +1,248 @@
+# **Troubleshoot Spark application**
+To obtain application error details, find the CloudWatch log or s3 log location defined by your job submission script. 
+For example the driver log is stored in s3://S3BUCKET_NAME/emr-containers-log/CLUSTER_ID/jobs/JOB_ID/containers/spark-JOB_ID/spark-JOB_ID-driver/stderr.gz defined by the following submission config:
+```bash
+--configuration-overrides '{
+    "monitoringConfiguration": {
+      "s3MonitoringConfiguration": {"logUri": "s3://'$S3BUCKET'/emr-containers-log"}}}'
+```
+
+##**Scenario 1 - "PersistentVolumeClaims is forbidden"**
+
+Running Spark jobs that require creation, listing or deletion of Persistent Volume Claims (PVC) is not supported before EMR6.8. Jobs that require these permissions will fail with the exception “persistentvolumeclaims is forbidden". Looking into application logs, you can see the driver log shows an error something look like this: `persistentvolumeclaims is forbidden. User "system:serviceaccount:emr:emr-containers-sa-spark-client-93ztm12rnjz163mt3rgdb3bjqxqfz1cgvqh1e9be6yr81" cannot create resource "persistentvolumeclaims" in API group "" in namesapce "emr".`
+
+###**Cause:** 
+This is due to the default Kubernetes role: “emr-containers” missing the required PVC permissions. As a result, the `emr-containers` primary role can’t dynamically create necessary permissions in its childern roles of Spark driver, Spark executor or Spark client when submit a job. 
+
+###**Solution:**
+Add the required permissions to emr-containers related roles in EKS.
+
+###**Verification**
+To verify whether your **emr-containers** role has the necessary permissions, run the command:
+```bash
+export NAMESPACE=YOUR_VALUE
+kubectl describe role emr-containers -n ${NAMESPACE}
+```
+In addition, verify whether the spark driver/client roles have the necessary permissions with:
+```bash
+kubectl describe role emr-containers-role-spark-driver -n ${NAMESPACE}
+kubectl describe role emr-containers-role-spark-client -n ${NAMESPACE}
+```
+If the permissions aren’t there, proceed with the **Patch** steps:
+
+###**Patch via automated script**
+
+1.If there’s any job(s) running that needs the PVC permissions and failing, stop the job(s).
+2.Create a script file named `RBAC_Patch.py`:
```

Comment:
Do you think its better to store in Github somewhere so that we keep solution small? Reason being, we will add multiple troubleshooting scenarios and would like to keep solution simple and short

Reply:
moved the python script a central place and shorten the troubleshoot section